src/ring_theory/mv_polynomial/basic.lean

```diff
@@ -117,3 +117,11 @@ rfl
 end degree
 
 end mv_polynomial
+
+/-- The monomials form a basis on `polynomial R`. -/
+noncomputable def polynomial.basis_monomials : basis ℕ R (polynomial R) :=
```

Comment:
Should this go in `ring_theory/polynomial/basic.lean` instead?

Reply:
I agree, but `finsupp.basis_single_one` is in `linear_algebra.finsupp_vector_space` that imports `data.mv_polynomial`, so I don't think it is possible without a rather big change.

Reply:
I don't follow your argument; what's the relevance of `data/mv_polynomial`?

Reply:
If I import `linear_algebra.finsupp_vector_space` in `ring_theory/polynomial/basic.lean` I get a circular import (well, I get tons of errors, usually this happens to me with circular import). I am not sure it starts at `data/mv_polynomial` or somewhere else

Reply:
It's `linear_algebra.finsupp_vector_space` --> `linear_algebra.finite_dimensional` --> `field_theory.finiteness` --> `ring_theory.finiteness` --> `ring_theory.algebra_tower` --> `ring_theory.adjoin.basic` --> `ring_theory.polynomial.basic`.

Reply:
Thanks for the full investigation! I agree that's not worth attempting to untangle, nor is it really worth creating a new file for two one-line results. Leaving these here is fine.